association.go

```diff
@@ -599,6 +683,82 @@ func (a *Association) handleReconfig(c *chunkReconfig) ([]*packet, error) {
 	return pp, nil
 }
 
+func (a *Association) handleForwardTSN(c *chunkForwardTSN) []*packet {
+	fmt.Printf("handleForward: %s\n", c.String())
+
+	if !a.useForwardTSN {
+		// Return an error chunk
+		cerr := &chunkError{
+			errorCauses: []errorCause{&errorCauseUnrecognizedChunkType{}},
+		}
+		outbound := &packet{}
+		outbound.verificationTag = a.peerVerificationTag
+		outbound.sourcePort = a.sourcePort
+		outbound.destinationPort = a.destinationPort
+		outbound.chunks = []chunk{cerr}
+		return []*packet{outbound}
+	}
+
+	// From RFC 3758 Sec 3.6:
+	//   the receiver MUST perform the same TSN handling, including duplicate
+	//   detection, gap detection, SACK generation, cumulative TSN
+	//   advancement, etc. as defined in RFC 2960 [2]---with the following
+	//   exceptions and additions.
+
+	//   When a FORWARD TSN chunk arrives, the data receiver MUST first update
+	//   its cumulative TSN point to the value carried in the FORWARD TSN
+	//   chunk,
+
+	// Advance peerLastTSN
+	for sna32LT(a.peerLastTSN, c.newCumulativeTSN) {
+		a.payloadQueue.pop(a.peerLastTSN + 1) // may not exist
+		a.peerLastTSN++
+	}
+
+	// From RFC 3758 Sec 3.6:
+	//   .. and then MUST further advance its cumulative TSN point locally
+	//   if possible
+	// Meaning, if peerLastTSN+1 points to a chunk that is received,
+	// advance peerLastTSN until peerLastTSN+1 points to unreceived chunk.
+	_, popOk := a.payloadQueue.pop(a.peerLastTSN + 1) // may not exist
```

Comment:
Could we combine these into one for? Incoming pseudo-code
```
for _, popOk = a.payloadQueue.pop(a.peerLastTSN + 1) {
	a.peerLastTSN++
}

Reply:
Agreed. Will do.

Reply:
Fixed @98bccc